src/library/list/LibraryItem.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+
+import { LIBRARY_TYPES, libraryShape } from '../common';
+import messages from './messages';
+
+const LibraryLink = ({type, url, children, ...attrs}) => {
+  if (type == LIBRARY_TYPES.LEGACY) {
+    return (
+      <a href={url} {...attrs}>
+        {children}
+      </a>
+    );
+  }
+
+  return (
+    <Link to={url} {...attrs}>
+      {children}
+    </Link>
+  );
+}
+
+LibraryLink.propTypes = {
+  type: PropTypes.string.isRequired,
+  url: PropTypes.string.isRequired,
+  children: PropTypes.node.isRequired,
+};
+
+/**
+ * Represents a library in the list of libraries.
+ */
+const LibraryItem = ({intl, library}) => (
```

Comment:
How about `LibraryListItem`? Might make it even clearer what this class is without having to read the above comment.

Reply:
Done!